tracer/test/Datadog.Trace.Tests/ManualInstrumentation/InstrumentationTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿// <copyright file="InstrumentationTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+extern alias DatadogTraceManual;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.ClrProfiler;
+using FluentAssertions;
+using Xunit;
+using ManualInstrumented = DatadogTraceManual::Datadog.Trace.SourceGenerators.InstrumentedAttribute;
+using ManualTracer = DatadogTraceManual::Datadog.Trace.Tracer;
+
+namespace Datadog.Trace.Tests.ManualInstrumentation;
+
+public abstract class InstrumentationTests<T>
+{
+    [Fact]
+    public void DatadogTraceManual_AllInstrumentedMethodsAreDecoratedWithInstrumentedAttribute()
```

Comment:
Would be nice if we could "snapshot" the duck type interfaces to avoid accidental version mismatch breakage :thinking: Would require additional annotations, but that's maybe ok, as I think we potentially need similar annotations for NativeAOT